### ISSUE TITLE
Incompatible loss values

### ISSUE THREAD

**Issue by vianamp** [kahn-jms/vae#1]
Hey James, I was looking for implementations of CVAE using Keras and TF2 and came across your code. Found it very nice! However, it seems that the values for reconstruction and KL losses are incompatible. By isolating each component I found that recloss is of order 1e5 and klloss is of order 1e-2. Have you notice the same problem?

**Comment by vianamp** [comment]
I think I found the problem. In your loss function I believe you want to calculate the average MSE. To do so you should replace `reconstruction_loss *= self.input_dim` with `reconstruction_loss /= self.input_dim`.
